### PR TITLE
feat(language-core): support `@vue-generic`

### PR DESCRIPTION
**Pull request by KazariEX**
resolve vuejs/core#8015

### CODE REVIEW

File: test-workspace/tsc/passedFixtures/vue3/v-generic/main.vue [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<script setup lang="ts">
+import { exactType } from '../../shared';
+import Comp from './comp.vue';
+</script>
+
+<template>
+    <Comp v-generic="number" @foo="(val) => exactType(val, {} as number)"/>
```
Comment (xiaoxiangmoe) [inline test-workspace/tsc/passedFixtures/vue3/v-generic/main.vue:7]: Maybe we should also add a example for multiple generic params

**Comment by johnsoncodehk** [comment]
I think writing types in `v-generic` is wrong in Vue's specification, according to the documentation custom directives only receive JS statements, the `v-generic` convention may break compatibility with other tools/libraries.

We should use `generic`, or convention to pass type parameters in HTML comment through `<!-- @vue-generic {xxx} -->`.

**Comment by johnsoncodehk** [comment]
@KazariEX I see, but this will still break Vue versions prior to https://github.com/vuejs/core/pull/12301, especially since Vue 2 has stopped being maintained. Since this is a language tools specific feature, we need an implementation that doesn't have to change core.

**Comment by KazariEX** [comment]
Can we consider it as a feature of the next minor version?

**Pull request merged**